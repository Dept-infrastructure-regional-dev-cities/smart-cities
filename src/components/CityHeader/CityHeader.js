import React from 'react';
import PropTypes from 'prop-types';
import getColorVariant from '../../helpers/getColorVariant';
import style from './CityHeader.scss';

const shades = [
  '500',
  '600',
  '700',
  '800',
  '900',
];

const CityHeader = (props) => {
  const backgroundPanels = shades.map(shade => (
    <div
      key={shade}
      className={style.backgroundPanel}
      style={{
        background: getColorVariant(props.categoryColorName, shade),
      }}
    />
  ));

  return (
    <div className={style.wrapper}>
      <div className={style.backgroundPanelWrapper}>
        {backgroundPanels}
      </div>

      <div className={style.container}>
        <h1 className={style.title}>
          {props.cityName}
        </h1>

        <span className={style.iconWrapper} />
      </div>
    </div>
  );
};

CityHeader.propTypes = {
  cityName: PropTypes.string.isRequired,
  categoryColorName: PropTypes.string.isRequired,
};

export default CityHeader;
