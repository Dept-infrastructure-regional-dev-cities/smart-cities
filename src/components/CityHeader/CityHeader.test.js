import React from 'react';
import { shallow } from 'enzyme';
import CityHeader from './CityHeader';

const defaultProps = {
  cityName: 'Greater perth',
  categoryColorName: 'jobs',
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityHeader {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
