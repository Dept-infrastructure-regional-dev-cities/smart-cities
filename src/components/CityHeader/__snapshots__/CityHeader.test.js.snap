// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <div className=\\"backgroundPanelWrapper\\">
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
  </div>
  <div className=\\"container\\">
    <h1 className=\\"title\\">
      Greater perth
    </h1>
    <span className=\\"iconWrapper\\" />
  </div>
</div>"
`;
